src/ripple/app/tx/impl/SusPay.cpp

```diff
@@ -183,23 +183,22 @@ SusPayCreate::doApply()
             return tecNO_PERMISSION;
     }
 
-    XRPAmount const amount =
-        STAmount(ctx_.tx[sfAmount]).mantissa();
-
     auto const account = ctx_.tx[sfAccount];
-
     auto const sle = ctx_.view().peek(
         keylet::account(account));
 
-    if (XRPAmount(STAmount((*sle)[sfBalance]).mantissa()) <
-        XRPAmount(ctx_.view().fees().accountReserve(
-            (*sle)[sfOwnerCount] + 1)))
-        return tecINSUFFICIENT_RESERVE;
+    // Check reserve and funds availability
+    {
+        auto const balance = STAmount((*sle)[sfBalance]).xrp();
```

Comment:
We could, but I wanted to leverage `XRPAmount` wherever possible and implicit conversions to `STAmount`